project/Tests/CoreTests/Utils/CoreRender+Test.swift

```diff
@@ -0,0 +1,37 @@
+//  Copyright © 2020 The nef Authors.
+
+import XCTest
+import NefModels
+import Bow
+import BowEffects
+
+
+extension XCTestCase {
+    func markdown(content: String) -> IO<CoreRenderError, RendererOutput<String>> {
+        CoreRender.markdown.render(content: content)
+                           .provide(CoreMarkdownEnvironment())
+    }
+    
+    func jekyll(content: String, permalink: String) -> IO<CoreRenderError, RendererOutput<String>> {
+        CoreRender.jekyll.render(content: content)
+                         .provide(CoreJekyllEnvironment(permalink: permalink))
+    }
+    
+    func carbon(content: String, downloader: CarbonDownloader, style: CarbonStyle) -> IO<CoreRenderError, RendererOutput<Image>> {
+        CoreRender.carbon.render(content: content)
+                         .provide(CoreCarbonEnvironment(downloader: downloader, style: style))
+    }
+}
+
+
+extension XCTestCase {
+    func assert<A: Equatable>(_ io: IO<CoreRenderError, RendererOutput<A>>, succeeds: A) {
```

Comment:
We should be able to pass line, file and info message to these assertions, with appropriate defaults, to ensure errors are reported in the correct place.